packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -136,39 +135,35 @@ class Auth {
             fetchOptions: config.fetchOptions
         })
 
-        if (this.get('site_id') && this.get('site_id') !== config.siteId) {
-            // if site is switched, remove all existing auth data in storage
-            // and the next auth.ready() call with restart the auth flow
-            this.clearStorage()
-            this.pendingToken = undefined
-        }
-
-        if (!this.get('site_id')) {
-            this.set('site_id', config.siteId, {
-                expires: this.REFRESH_TOKEN_EXPIRATION_DAYS
-            })
-        }
+        // NOTE: Not too sure how I feel about this, seems a little confusing. Should
+        // the DATA_MAP instead have definitions for serverStorage and clientStorage?
+        // Another idea is to implement a getPrefix function on the storage class. But that
+        // would look kinda gross too.
+        const storageOptions = {keyPrefix: config.siteId}
+        this.stores = onClient
+            ? {
+                  cookie: new CookieStorage(storageOptions),
+                  local: new LocalStorage(storageOptions)
+              }
+            : {
+                  cookie: new ServerStorage(storageOptions),
+                  local: new ServerStorage(storageOptions)
+              }
```

Comment:
I wanna call out and discuss the way we initialize the storages...

Prior to this change, we initialize the storage outside of the `Auth` class, globally, and the reason for that is we want to the token to persist on server side across lambda invocations. As we all know, two serial SLAS calls for every single visit on the server side is expensive. (I also noticed a bug with the previous implementation, that the storage won't properly refresh when config value change, except `siteId`)

After this change, the storages are bound to the instance of the Auth class. Right now, we re-initialize Auth class in the provider when config changes.

```jsx
// We do this in provider
const auth = useMemo(() => {
        return new Auth({
            clientId,
            organizationId,
            shortCode,
            siteId,
            proxy,
            redirectURI,
            fetchOptions
        })
    }, [clientId, organizationId, shortCode, siteId, proxy, redirectURI, fetchOptions])
```

When a new Auth instance is created, the storage got wiped. I think this is expected behavior and fixed the bug i mentioned above. But we lose the ability to cache token on lambda, what can we do to keep caching token on lambda? 🤔

Reply:
So to maintain the desired behaviour of retaining the token storage on the server side between lambda runs, I've added the ability for the MemoryStorage class to use a global map if required (which it is for our case), I called this option `sharedContext` so that one Auth class instance will use the same data storage as another Auth class instance.